### ISSUE TITLE
Laravel 11

### ISSUE THREAD

**Issue by sbc640964** [tenancy/tenancy#293]
We are waiting for an upgrade - compatibility with 11 :)

**Comment by ArlonAntonius** [comment]
It's coming, I need to make some small tweaks tonight to make sure everything is okay :)

For anyone interested, changes are being prepared in #289

**Comment by dligero-mdmsystems** [comment]
Hi guys! 

Any advance on this topic? 
It would be nice to get full support of Laravel 11.

Thank you so much!

**Comment by shafimsp** [comment]
Any update on Laravel 11 compatibility?..

**Comment by ArlonAntonius** [comment]
This has taken wayyyyy longer than it should have as we were trying to keep support for old versions and make sure new versions also work as expected. I just tagged [v2.2.0-beta2](https://github.com/tenancy/tenancy/releases/tag/v2.2.0-beta2), if you could also check if that works for you guys and it is not causing any problems, then I can tag a real version.
Testing it myself as well :)

**Comment by thumbtech** [comment]
Well, I assume you are referring to v2.3.0-beta2 for testing L11 ... ?

Update to `src/Identification/Http/Providers/IdentificationProvider.php` from

`$this->app->extend(Kernel::class ...`

to

`$this->app->afterResolving(Kernel::class ...`

has broken the resolver there.

**Comment by ArlonAntonius** [comment]
@thumbtech Yeah I meant https://github.com/tenancy/tenancy/releases/tag/v2.3.0-beta2, thanks for correctign me.

What Laravel version is this on?

**Comment by thumbtech** [comment]
v11.4.0:
```
sail composer info
brick/math                         0.11.0       Arbitrary-precision arithmetic library
carbonphp/carbon-doctrine-types    3.2.0        Types to use Carbon in Doctrine
dflydev/dot-access-data            v3.0.2       Given a deep data structure, access data by dot notation.
doctrine/inflector                 2.0.10       PHP Doctrine Inflector is a small library that can perform str...
doctrine/lexer                     3.0.1        PHP Doctrine Lexer parser library that can be used in Top-Down...
dragonmantank/cron-expression      v3.3.3       CRON for PHP: Calculate the next or previous run date and dete...
egulias/email-validator            4.0.2        A library for validating emails against several RFCs
fakerphp/faker                     v1.23.1      Faker is a PHP library that generates fake data for you.
filp/whoops                        2.15.4       php error handling for cool kids
fruitcake/php-cors                 v1.3.0       Cross-origin resource sharing library for the Symfony HttpFoun...
graham-campbell/result-type        v1.1.2       An Implementation Of The Result Type
guzzlehttp/guzzle                  7.8.1        Guzzle is a PHP HTTP client library
guzzlehttp/promises                2.0.2        Guzzle promises library
guzzlehttp/psr7                    2.6.2        PSR-7 message implementation that also provides common utility...
guzzlehttp/uri-template            v1.0.3       A polyfill class for uri_template of PHP
hamcrest/hamcrest-php              v2.0.1       This is the PHP port of Hamcrest Matchers
laravel/framework                  v11.4.0      The Laravel Framework.
laravel/pint                       v1.15.1      An opinionated code formatter for PHP.
laravel/prompts                    v0.1.19
laravel/sail                       v1.29.1      Docker files for running a basic Laravel application.
laravel/serializable-closure       v1.3.3       Laravel Serializable Closure provides an easy and secure way t...
laravel/tinker                     v2.9.0       Powerful REPL for the Laravel framework.
league/commonmark                  2.4.2        Highly-extensible PHP Markdown parser which fully supports the...
league/config                      v1.2.0       Define configuration arrays with strict schemas and access val...
league/flysystem                   3.27.0       File storage abstraction for PHP
league/flysystem-local             3.25.1       Local filesystem adapter for Flysystem.
league/mime-type-detection         1.15.0       Mime-type detection for Flysystem
mockery/mockery                    1.6.11       Mockery is a simple yet flexible PHP mock object framework
monolog/monolog                    3.6.0        Sends your logs to files, sockets, inboxes, databases and vari...
myclabs/deep-copy                  1.11.1       Create deep copies (clones) of your objects
nesbot/carbon                      3.2.4        An API extension for DateTime that supports 281 different lang...
nette/schema                       v1.3.0       📐 Nette Schema: validating data structures against a given ...
nette/utils                        v4.0.4       🛠  Nette Utils: lightweight utilities for string & array ma...
nikic/php-parser                   v5.0.2       A PHP parser written in PHP
nunomaduro/collision               v8.1.1       Cli error handling for console/command-line PHP applications.
nunomaduro/termwind                v2.0.1       Its like Tailwind CSS, but for the console.
phar-io/manifest                   2.0.4        Component for reading phar.io manifest information from a PHP ...
phar-io/version                    3.2.1        Library for handling version information and constraints
phpoption/phpoption                1.9.2        Option Type for PHP
phpunit/php-code-coverage          11.0.3       Library that provides collection, processing, and rendering fu...
phpunit/php-file-iterator          5.0.0        FilterIterator implementation that filters files based on a li...
phpunit/php-invoker                5.0.0        Invoke callables with a timeout
phpunit/php-text-template          4.0.0        Simple template engine.
phpunit/php-timer                  7.0.0        Utility class for timing
phpunit/phpunit                    11.1.2       The PHP Unit Testing framework.
psr/clock                          1.0.0        Common interface for reading the clock.
psr/container                      2.0.2        Common Container Interface (PHP FIG PSR-11)
psr/event-dispatcher               1.0.0        Standard interfaces for event handling.
psr/http-client                    1.0.3        Common interface for HTTP clients
psr/http-factory                   1.0.2        Common interfaces for PSR-7 HTTP message factories
psr/http-message                   2.0          Common interface for HTTP messages
psr/log                            3.0.0        Common interface for logging libraries
psr/simple-cache                   3.0.0        Common interfaces for simple caching
psy/psysh                          v0.12.3      An interactive shell for modern PHP.
ralouphie/getallheaders            3.0.3        A polyfill for getallheaders.
ramsey/collection                  2.0.0        A PHP library for representing and manipulating collections.
ramsey/uuid                        4.7.5        A PHP library for generating and working with universally uniq...
sebastian/cli-parser               3.0.1        Library for parsing CLI options
sebastian/code-unit                3.0.0        Collection of value objects that represent the PHP code units
sebastian/code-unit-reverse-lookup 4.0.0        Looks up which function or method a line of code belongs to
sebastian/comparator               6.0.0        Provides the functionality to compare PHP values for equality
sebastian/complexity               4.0.0        Library for calculating the complexity of PHP code units
sebastian/diff                     6.0.1        Diff implementation
sebastian/environment              7.1.0        Provides functionality to handle HHVM/PHP environments
sebastian/exporter                 6.0.1        Provides the functionality to export PHP variables for visuali...
sebastian/global-state             7.0.1        Snapshotting of global state
sebastian/lines-of-code            3.0.0        Library for counting the lines of code in PHP source code
sebastian/object-enumerator        6.0.0        Traverses array structures and object graphs to enumerate all ...
sebastian/object-reflector         4.0.0        Allows reflection of object attributes, including inherited an...
sebastian/recursion-context        6.0.0        Provides functionality to recursively process PHP variables
sebastian/type                     5.0.0        Collection of value objects that represent the types of the PH...
sebastian/version                  5.0.0        Library that helps with managing the version number of Git-hos...
spatie/backtrace                   1.5.3        A better backtrace
spatie/flare-client-php            1.4.4        Send PHP errors to Flare
spatie/ignition                    1.13.2       A beautiful error page for PHP applications.
spatie/laravel-ignition            2.5.2        A beautiful error page for Laravel applications.
symfony/clock                      v7.0.5       Decouples applications from the system clock
symfony/console                    v7.0.6       Eases the creation of beautiful and testable command line inte...
symfony/css-selector               v7.0.3       Converts CSS selectors to XPath expressions
symfony/deprecation-contracts      v3.4.0       A generic function and convention to trigger deprecation notices
symfony/error-handler              v7.0.6       Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher           v7.0.3       Provides tools that allow your application components to commu...
symfony/event-dispatcher-contracts v3.4.2       Generic abstractions related to dispatching event
symfony/finder                     v7.0.0       Finds files and directories via an intuitive fluent interface
symfony/http-foundation            v7.0.6       Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                v7.0.6       Provides a structured process for converting a Request into a ...
symfony/mailer                     v7.0.6       Helps sending emails
symfony/mime                       v7.0.6       Allows manipulating MIME messages
symfony/polyfill-ctype             v1.29.0      Symfony polyfill for ctype functions
symfony/polyfill-intl-grapheme     v1.29.0      Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-idn          v1.29.0      Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 funct...
symfony/polyfill-intl-normalizer   v1.29.0      Symfony polyfill for intl's Normalizer class and related funct...
symfony/polyfill-mbstring          v1.29.0      Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.29.0      Symfony polyfill backporting some PHP 7.2+ features to lower P...
symfony/polyfill-php80             v1.29.0      Symfony polyfill backporting some PHP 8.0+ features to lower P...
symfony/polyfill-php83             v1.29.0      Symfony polyfill backporting some PHP 8.3+ features to lower P...
symfony/polyfill-uuid              v1.29.0      Symfony polyfill for uuid functions
symfony/process                    v7.0.4       Executes commands in sub-processes
symfony/routing                    v7.0.6       Maps an HTTP request to a set of configuration variables
symfony/service-contracts          v3.4.2       Generic abstractions related to writing services
symfony/string                     v7.0.4       Provides an object-oriented API to strings and deals with byte...
symfony/translation                v7.0.4       Provides tools to internationalize your application
symfony/translation-contracts      v3.4.2       Generic abstractions related to translation
symfony/uid                        v7.0.3       Provides an object-oriented API to generate and represent UIDs
symfony/var-dumper                 v7.0.6       Provides mechanisms for walking through any arbitrary PHP vari...
symfony/yaml                       v7.0.3       Loads and dumps YAML files
tenancy/framework                  v2.3.0-beta2 Creating multi tenant saas from your Laravel app with ease
tenancy/identification-driver-http v2.3.0-beta2 The tenancy/tenancy identification driver for http
theseer/tokenizer                  1.2.3        A small library for converting tokenized PHP source code into ...
tijsverkoyen/css-to-inline-styles  v2.2.7       CssToInlineStyles is a class that enables you to convert HTML-...
vlucas/phpdotenv                   v5.6.0       Loads environment variables from `.env` to `getenv()`, `$_ENV`...
voku/portable-ascii                2.0.1        Portable ASCII library - performance optimized (ascii) string ...
webmozart/assert                   1.11.0       Assertions to validate method input/output with nice error mes...
```

**Comment by sbc640964** [comment]
@ArlonAntonius,   is it work?

**Comment by sbc640964** [comment]
@ArlonAntonius 
![image](https://github.com/tenancy/tenancy/assets/46646044/6977170b-b91b-4722-8a53-a80f4a6313a0)

**Comment by sbc640964** [comment]
Anyone, Anything?
I hope you response!

**Comment by thumbtech** [comment]
@sbc640964 how are you using the package in your application? The package does a lot (perhaps too much)!

**Comment by sbc640964** [comment]
@thumbtech, Maybe I didn't understand your question.

At this point I'm just trying to install the package in laravel 11 (windows 11)
`composer require tenancy/tenancy`

**Comment by sbc640964** [comment]
@ArlonAntonius  

In the end, I installed the package as a local package, everything seems to work except the detection itself, I try to detect with an HTTP driver, and it does not detect on its own.

When used with `Tenancy::identifyTenant()` it is an excellent identifier.

**Comment by thumbtech** [comment]
@sbc640964 an update to the code noted in comment above has broken the HTTP identifier: https://github.com/tenancy/tenancy/issues/293#issuecomment-2067582645
As one of the most basic core features, having it break like this, even in beta, was concerning to say the least!

I'm not sure what plans are for the package, but given breadth of functionality, I think a new major release is in order with only L11+ support. Too much happening to support L11 and prior version IMO. I suppose this will be a hassle to support multiple releases going forward, though ... ? Given pace of Laravel major releases, I really feel for those devs supporting packages like this. My L11 upgrade was rough. It had quite a few issues when running in "L10 mode" as [advertised](https://laravel.com/docs/11.x/upgrade#application-structure). After converting into L11 app structure and with appropriate tweaks to code, it is great and I see why this is a better system moving forward.

FWIW I was forced to move on because of need for L11 support and have had good experience with the Spatie multitenancy package, which is narrower in scope but covered my needs. HTTP identification works well there and I just leveraged my code from this package to get it up and running.

**Comment by ArlonAntonius** [comment]
This took longer than expected and due to the many changes also in how Laravel boots its Application and Kernel now, it was difficult to keep compatibility with other versions.

We thought we had the right implementation before, but simply due to an oversight of a change in the kernel, we made a wrong adjustment.

We're working on 3.x for Tenancy with a lot more features, that is the version in which we will also start dropping older versions of Laravel, for now we wanted to keep the compatibility.

I'm testing the last functionality was we speak, release will happen in the next 48 hours.

**Comment by ArlonAntonius** [comment]
Released as v2.3.0, sorry for all the delays.
